retry/lib/retry.dart

```diff
@@ -122,13 +123,22 @@ class RetryOptions {
     FutureOr<T> Function() fn, {
     FutureOr<bool> Function(Exception) retryIf,
     FutureOr<void> Function(Exception) onRetry,
+    FutureOr<bool> Function(dynamic) retryIfValue,
+    FutureOr<void> Function(dynamic) onRetryIfValue,
```

Comment:
```suggestion
    FutureOr<void> Function(T) onRetryValue,
```
Why the if here? `onRetry` doesn't have that...

Also why are these `dynamic`, isn't `T` a better type? Will that work with `void`?